### PR TITLE
Adds the setup command

### PR DESCRIPTION
**Pull request by MichaelBaker**
I've added a little system for adding commands as well as the 'setup'
command which installs pathogen, creates our vpmrc file, and sets up the
user's vimrc file.

### CODE REVIEW

File: lib/vpm/commands.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module Commands
```
Comment (owenthereal) [inline lib/vpm/commands.rb:1]: should we namespace it with VPM?

File: lib/vpm/commands/install_all_plugins.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module Commands::InstallAllPlugins
+  def self.call(args)
+    vim_plugins_file = File.join(File.expand_path('.'), 'VimPlugins')
+    content          = File.read(vim_plugins_file)
+    plugins          = VPM::ManifestParser.parse(content)
+    plugins.each(&:install)
+  end
+end
```
Comment (owenthereal) [inline lib/vpm/commands/install_all_plugins.rb:8]: It's prob. be better part of `vpm install` as we discussed.

File: lib/vpm/commands/invalid_command.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Commands::InvalidCommand
+  def initialize(invalid_command)
+    @invalid_command = invalid_command
+  end
+
+  def call(args)
+    puts "'#{@invalid_command}' is not a valid command."
+    exit 1
+  end
+end
```
Comment (owenthereal) [inline lib/vpm/commands/invalid_command.rb:10]: +1 for NullObject

File: lib/transaction.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      unless task.perform
+        completed_tasks.each &:undo
+        break
+      end
+    end
+  rescue Exception => e
+    completed_tasks.each &:undo
+    raise
+  end
+end
```
Comment (owenthereal) [inline lib/transaction.rb:27]: +1 for Transaction to manage running of each setup of a command

**Comment by owenthereal** [comment]
Generally look good to me. We need to come up with a way to merge the two branches. Thoughts?

**Comment by MichaelBaker** [comment]
Get together after work or on Saturday and go over them?

File: lib/vpm/commands.rb (hunk 1 above)
Comment (MichaelBaker) [inline lib/vpm/commands.rb:1]: We can, we might end up namespacing every class in our app though :^P

File: lib/vpm/commands/install_all_plugins.rb (hunk 2 above)
Comment (MichaelBaker) [inline lib/vpm/commands/install_all_plugins.rb:8]: I agree

File: lib/vpm/commands.rb (hunk 1 above)
Comment (owenthereal) [inline lib/vpm/commands.rb:1]: Yup, that's how people draw a line between their gems and others' :)

**Comment by owenthereal** [comment]
Let's do it after work tomorrow, I will merge one branch into master first and we will go over and fix the conflicts for the other.

File: lib/vpm/commands.rb (hunk 1 above)
Comment (MichaelBaker) [inline lib/vpm/commands.rb:1]: Ok, I'll go through and add our namespace to the commands and tasks tonight.

**Pull request closed without merging**